pkg/eventstore/types.go

```diff
@@ -13,6 +13,10 @@ const (
 
 type Store interface {
 	Bucket(name string) (Bucket, error)
+
+	// LoadBucketWithNoPurge loads the bucket with no purge run.
+	// This is useful for loading the bucket for read-only operations.
+	LoadBucketWithNoPurge(name string) (Bucket, error)
```

Comment:
It's better to use options pattern for `Bucket` method, therefore we don't need to add the new method.. and it's unreasonable.